app/routes.py

```diff
@@ -1,2 +1,289 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, request
+import requests
+from datetime import datetime
+import os
+from dotenv import load_dotenv
 
+from tests.test_wave_06 import test_post_task_ids_to_goal
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+
+    new_task = Task(title=request_body["title"],
+                    description=request_body["description"],
+                    completed_at=request_body["completed_at"])
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    response_body = {
+        "task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": new_task.completed_at is not None
+        }
+    }
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])
+def read_task():
+
+    # tasks hold all the information from the database. Becasue Task is referring to a table. So we are capturing all the records(rows) in the table and holding it in tasks.
+    # queries are the records (i.e. rows)
+    # creating task_response variable
+    title_query = request.args.get("sort")
+    if title_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif title_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+
+    task_responses = []
+    for task in tasks:
+        task_response = {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+        if task.goal_id:
+            task_response["goal_id"] = task.goal_id
+        task_responses.append(task_response)
+    return jsonify(task_responses), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    task_response = {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": task.completed_at is not None
+    }
+    if task.goal_id:
+        task_response["goal_id"] = task.goal_id
+
+    response_body = {"task": task_response}
+
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
We should be sure that the same fields required for POSTing are included here for PUT. PUT _replaces_ the value for the supplied task id, so we should ensure that all of the values required to represent a `Task` are provided with the same restrictions as we had when creating it in the first place.

At the least, we should ensure that the keys are present in the dictionary before trying to access them, otherwise this request would cause a `KeyError`.